### DISCUSSION TITLE
How to generate ground truth labels for semantic segmentation from json file

### DISCUSSION THREAD

**Discussion by rod409** [bdd100k/bdd100k#77] [Q&A] [answered]
I wanted to generate label masks from the json files for semantic segmentation. I tried running the command below following the documentation to get instance segmentation masks but all the output images have only one value for every pixel. What is the right process to get semantic segmentation ground truths from the JSON file? Also, do I need to convert instance to semantic segmentation or is there code to directly generate semantic masks?

`python3 -m bdd100k.vis.labels --image-dir bdd100k/images/10k/val -l bdd100k/labels/bdd100k_labels_images_val.json -s 1
--instance -o bdd100k/mask/val/`

**Comment by XiaLiPKU** [comment, marked as answer]
Thanks for informing us!
We just update the scripts these days, now you can follow the command there: https://doc.bdd100k.com/format.html#to-bitmasks
For the definition of `bitmask`, you may refer to here: https://doc.bdd100k.com/format.html#segmentation-format
Most importantly, we have updated download files, after unzip them, you can already get bitmask as described in: https://doc.bdd100k.com/download.html#semantic-segmentation

Above is for semantic segmentation. For instance segmentation, you can download files as: https://doc.bdd100k.com/download.html#instance-segmentation

**Comment by gwc2017180303** [comment]
This command requires too much to input  and it is easy to make mistakes. Can the script be changed like cityscapes.

**Reply by XiaLiPKU** [reply]
> This command requires too much to input and it is easy to make mistakes. Can the script be changed like cityscapes.

Reasonable suggestion. We are now considering using the traditional format: one-channel png as the label.

**Reply by XiaLiPKU** [reply]
Now we have updated the evaluation script and doc, please refer here: https://doc.bdd100k.com/evaluate.html#id3
If the page is not updated, please use Ctrl + F5 in your browser to re-load all resources. Hope this can help you
